spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -163,8 +163,12 @@ else if (this.listenerConsumer.assignedPartitions != null) {
 			for (Map.Entry<MetricName, ? extends Metric> entry : listenerConsumer.consumer.metrics().entrySet()) {
 				MetricName metricName = entry.getKey();
 				Metric metric = entry.getValue();
-				String client = metricName.tags().get("client-id");
-				Map<MetricName, Metric> clientMetrics = metrics.computeIfAbsent(client, k -> new HashMap<>());
+				String clientId = metricName.tags().get("client-id");
+				Map<MetricName, Metric> clientMetrics = metrics.get(clientId);
+				if (clientMetrics == null) {
```

Comment:
Well, @garyrussell 's idea was to return here `Collections.singletonMap()`. So, the logic would be a bit different in this method.
Also I would like to see `Collections.unmodifableMap()` in the `ConcurrentMessageListenerContainer`. 
We really should not allow end-user to modify the result of the getter...